cheroot/test/test_server.py

```diff
@@ -236,3 +237,51 @@ def test_peercreds_unix_sock_with_lookup(peercreds_enabled_server):
         peercreds_text_resp = requests.get(unix_base_uri + PEERCRED_TEXTS_URI)
         peercreds_text_resp.raise_for_status()
         assert peercreds_text_resp.text == expected_textcreds
+
+
+def test_high_number_of_file_descriptors():
+    """Test the server does not crash with a high file-descriptor value.
+
+    This test should cause a server crash, as the server will try to
+    select() a file-descriptor higher than 1024.
+    """
+    # Get current resource limits to restore them later
+    (soft_limit, hard_limit) = resource.getrlimit(resource.RLIMIT_NOFILE)
+    increased_nofile_limit = 2048
+
+    # Create our server
+    httpserver = HTTPServer(
+        bind_addr=(ANY_INTERFACE_IPV4, EPHEMERAL_PORT), gateway=Gateway,
+    )
+    # Hoard a lot of file descriptors by opening and storing a lot of sockets
+    test_sockets = []
+
+    try:
+        # We have to increase the nofile limit above 1024
+        # Otherwise we see a 'Too many files open' error, instead of
+        # an error due to the file descriptor number being too high
+        resource.setrlimit(
+            resource.RLIMIT_NOFILE,
+            (increased_nofile_limit, increased_nofile_limit),
+        )
+
+        # Open a lot of file descriptors, so the next ones the server
+        # opens are higher than 1024
+        for i in range(1024):
+            test_sockets.append(socket.socket())
+
+        httpserver.prepare()
+        # This will trigger a crash if select() is used in the implementation
+        httpserver.tick()
```

Comment:
Let's add some assertions (it's a test after all)
```suggestion
        assert sockets[-1].fileno() >= 1024
        # This will trigger a crash if select() is used in the implementation
        httpserver.tick()
        with socket.socket() as sock:
            assert sock.fileno() >= 1024
```

Reply:
Updated, good idea

Reply:
FYI it's a good tone to accept suggested changes via GH UI because it preserves the patch authorship.